components/service/fretboard/README.md

```diff
@@ -12,6 +12,213 @@ Use gradle to download the library from JCenter:
 implementation "org.mozilla.components:fretboard:{latest-version}
 ```
 
+### Filters
+Fretboard allows you to specify the following filters:
+- Buckets: Every user is in one of 100 buckets (0-99). For every experiment you can set up a min and max value (0 <= min <= max <= 100). The bounds are [min, max).
+
+- appId (regex): The app ID (package name)
+- version (regex): The app version
+- country (regex): country, pulled from the default locale
+- lang (regex): language, pulled from the default locale
+- device (regex): Android device name
+- manufacturer (regex): Android device manufacturer
+- region: custom region, different from the one from the default locale (like a GeoIP, or something similar).
+For this to work you must provide a RegionProvider implementation when creating the Fretboard instance.
+
+### Creating Fretboard instance
+In order to use the library, first you have to create a new Fretboard instance. You do this once per app launch 
+(typically in your Application class onCreate method). You simply have to instantiate the Fretboard class and
+provide the ExperimentStorage and ExperimentSource implementations, and then call the loadExperiments method
+to load experiments from local storage:
```

Comment:
Let's move this up. Let's try to structure it in a way from "Where do I get the library? -> How do I initialize it? How do I fetch the configuration / update it -> How do I know if I am in an experiment -> How do I get the metadata?" ... then continue with the details about filters and how to customize functionality.